modal/cls.py

```diff
@@ -161,45 +161,60 @@ def __init__(
             check_valid_cls_constructor_arg(i + 1, arg)
         for key, kwarg in kwargs.items():
             check_valid_cls_constructor_arg(key, kwarg)
-
-        self._method_functions = {}
```

Comment:
Instead of eagerly constructing these lazy _Functions on class instantiation, we now conditionally run the same logic for each method in the `__getattr__` of Obj below, when users attempt ot access attribute or method of an instance.
This means we can now construct an `_Obj` without having method definitions of the class through local code or hydration metadata, as long as we have a `_Cls` object which itself is lazily loadable later on, when a user calls .remote()